### PR TITLE
fix(images): fix multi upload bug

### PR DESCRIPTION
**Pull request by Jekrimo**
**What this PR does** 📖
Fix multi upload bug

**Which issue(s) this PR fixes** 🔨
AP-274

**Additional comments** 🎤
Not nearly as pretty of code but is fully functional for me!

### CODE REVIEW

File: components/views/files/upload/Upload.vue [hunk 1]
```diff
@@ -152,28 +154,74 @@ export default Vue.extend({
...
+            //   this.cancelUpload()
+            //   document.body.style.cursor= PropCommonEnum.DEFAULT
+            //   this.$store.dispatch('textile/clearUploadStatus')
+            //   this.disabledButton = false
+            // })
+          // }, Promise.resolve().then(() => {
+          //   this.cancelUpload()
+          //   document.body.style.cursor= PropCommonEnum.DEFAULT
+          //   this.$store.dispatch('textile/clearUploadStatus')
+          //   this.disabledButton = false
```
Comment (stavares843) [inline components/views/files/upload/Upload.vue:223]: should we burn the commented-out code? 🔨

Comment (Jekrimo) [reply components/views/files/upload/Upload.vue]: Yup! Good catch thank you!

File: store/textile/actions.ts [hunk 2]
```diff
@@ -271,6 +270,8 @@ export default {
           },
         )
 
+    // $TextileManager.bucketManager?.addToIndex(file.file, result?.root, path)
```
Comment (stavares843) [inline store/textile/actions.ts:273]: one more here 🔨

**Review by stavares843 — CHANGES_REQUESTED** [review]
multi-upload works ✅ 

but takes some time to send out the images, they are 100kb or so

https://user-images.githubusercontent.com/29093946/147042425-5006d0aa-b8b5-44e1-bd77-4fcd3a1d3939.mp4

and if we can fix the glitch 
https://github.com/Satellite-im/Core-PWA/pull/703#issuecomment-998264410

it was fixed on the previous PR before merging but not sure why after merge, the bug appeared

**Comment by Jekrimo** [comment]
> multi-upload works ✅
> 
> but takes some time to send out the images, they are 100kb or so
> 
>  Screen.Recording.2021-12-22.at.5.50.04.AM.mp4 
> and if we can fix the glitch [#703 (comment)](https://github.com/Satellite-im/Core-PWA/pull/703#issuecomment-998264410)
> 
> it was fixed on the previous PR before merging but not sure why after merge, the bug appeared
> 
> Chrome, macOS

Hey, is this comment done? I'm not sure what the other bug is?

**Comment by Jekrimo** [comment]
> multi-upload works ✅
> 
> but takes some time to send out the images, they are 100kb or so
> 
>  Screen.Recording.2021-12-22.at.5.50.04.AM.mp4 
> and if we can fix the glitch [#703 (comment)](https://github.com/Satellite-im/Core-PWA/pull/703#issuecomment-998264410)
> 
> it was fixed on the previous PR before merging but not sure why after merge, the bug appeared
> 
> Chrome, macOS

Should be good now!

**Comment by stavares843** [comment]
The glitch is fixed, thanks! 🎉 

do you know if we can improve the speed when sending out the images?

**Comment by stavares843** [comment]
Captura de ecrã 2021-12-22, às 16 22 45


seems is no longer sending all now, but the glitch is fixed 🔨

File: components/views/chat/chatbar/Chatbar.vue [hunk 3]
```diff
@@ -302,7 +302,7 @@ export default Vue.extend({
      */
     handleDrop(e: any) {
```
Comment (WanderingHogan) [inline components/views/chat/chatbar/Chatbar.vue:303]: I know you didn't add this but can you change this from 'any' to DragEvent, and put the handleUpload into an if, like `if(e.dataTransfer){`

File: components/views/chat/chatbar/Chatbar.vue [hunk 4]
```diff
@@ -314,20 +314,21 @@ export default Vue.extend({
       /* Don't use event.preventDefault(). It prevent original text copy-paste */
       e.stopPropagation()
```
Comment (WanderingHogan) [inline components/views/chat/chatbar/Chatbar.vue:315]: Can you change the handlePaste from any to ClipboardEvent, same 'if' change as above

File: components/views/chat/chatbar/Chatbar.vue [hunk 5]
```diff
@@ -314,20 +314,21 @@ export default Vue.extend({
...
+      this.handleUpload(e.clipboardData.items, e)
     },
     /**
      * @method handleUpload
      * @description Takes in an array of event items and uploads the file objects
      * @param items Array of objects
      * @example this.handleUpload(someEvent.itsData.items)
      */
-    handleUpload(items: Array<object>) {
+    handleUpload(items: Array<object>, e: any) {
```
Comment (WanderingHogan) [inline components/views/chat/chatbar/Chatbar.vue:325]: I think this would just be 'Event'? But I am not sure

**Comment by stavares843** [comment]
https://user-images.githubusercontent.com/29093946/147142874-302367d5-2f79-4c23-b36c-c8c0886832e9.mp4

neat! 🎉

**Review by stavares843 — APPROVED** [review]

**Comment by stavares843** [comment]
to be done later - [AP-339](https://satellite-im.atlassian.net/browse/AP-339)

thanks for adding the ticket @Jekrimo 🎉

File: libraries/Textile/BucketManager.ts [hunk 6]
```diff
@@ -147,11 +147,11 @@ export default class BucketManager {
...
-  async removeFile(file: File, path: string) {
-    if (!this.buckets || !this.bucketKey) return;
-    this.buckets.removePath(this.bucketKey, `${this.prefix}${path}`);
-    // this.removeFromIndex(file);
-  }
+  // async removeFile(file: File, path: string) {
+  //   if (!this.buckets || !this.bucketKey) return;
+  //   this.buckets.removePath(this.bucketKey, `${this.prefix}${path}`);
+  //   // this.removeFromIndex(file);
+  // }
```
Comment (stavares843) [inline libraries/Textile/BucketManager.ts:154]: commented-out code 🔨

File: components/views/chat/chatbar/Chatbar.vue [hunk 7]
```diff
@@ -255,11 +256,10 @@ export default Vue.extend({
       sel?.selectAllChildren(messageBox)
       sel?.collapseToEnd()
 
-      if (this.$props.recipient) {
-        this.setChatText = {
-          userId: this.$props.recipient.address,
-          value: messageBox.innerHTML,
-        }
+      // if (messageBox.)
```
Comment (stavares843) [inline components/views/chat/chatbar/Chatbar.vue:259]: commented-out code 🔨

**Review by WanderingHogan — APPROVED** [review]

**Comment by stavares843** [comment]
@Jekrimo feel free to burn the commented out code on a next PR 🔨

**Pull request merged**